src/components/nav-item/nav-item.tsx

```diff
@@ -1,13 +1,28 @@
-export type NavLocationsProps = {
+import clsx from 'clsx';
+import {Link} from 'react-router-dom';
+import {AppRoute} from '@/constants';
+
+interface NavItemProps {
   location: string;
+  isActive: boolean;
+  onCityChangeClick: (city: string) => void;
 }
 
-export default function NavItem({location}: NavLocationsProps): JSX.Element {
+export default function NavItem({location, isActive, onCityChangeClick}: NavItemProps): JSX.Element {
   return (
     <li className="locations__item">
-      <a className="locations__item-link tabs__item" href="#">
+
+      <Link to={AppRoute.Root}
+        className={clsx(
+          'locations__item-link tabs__item',
+          isActive && 'locations__item-link tabs__item tabs__item--active'
+        )}
+        onClick={
+          (evt) => onCityChangeClick(evt.currentTarget.textContent || '')
```

Comment:
evt.currentTarget.textContent  надо забыть в принципе этот подход, как минимум у тебя может быть несколько языков на сайте из за чего он полностью не состоятелен. У тебя уже есть в компоненте location поэтому просто
```suggestion
          () => onCityChangeClick(location)
```